src/Rings/mpoly.jl

```diff
@@ -311,7 +311,14 @@ function singular_generators(B::IdealGens, monorder::MonomialOrdering=default_or
   singular_assure(B)
   # in case of quotient rings, monomial ordering is ignored so far in singular_poly_ring
   isa(B.gens.Ox, MPolyQuoRing) && return B.gens.S
-  isdefined(B, :ord) && B.ord == monorder && monomial_ordering(B.Ox, Singular.ordering(base_ring(B.S))) == B.ord && return B.gens.S
+  if isdefined(B, :ord) && (B.ord === monorder || B.ord == monorder)
+    # Whenever both are set, the monomial orderings on the Oscar side and the Singular side should agree.
+    # However, it might be the case that some (newly added) code is breaking this. In order to snoop 
+    # for that, we allow for the assertion level to be put up and then this will throw. 
+    @hassert :IdealGens 1 monomial_ordering(B.Ox, Singular.ordering(base_ring(B.S))) == B.ord
+    @assert monomial_ordering(B.Ox, Singular.ordering(base_ring(B.S))) == B.ord "orderings did not agree on the Oscar and the Singular side"
+    return B.gens.S
+  end
```

Comment:
My (unqualified) take on this would be to remove this whole `if` block. The `singular_assure` creates the ring on the Singular side with the same ordering as the internal ordering of the base ring on the OSCAR side. As long as one doesn't do anything fancy, this will be lex. (`B` has a field `keep_ordering` which appears to be `true` by default, maybe it should be `false`, no idea. And there are supposedly situations where the Singular ring is set from the beginning, no idea again.)
Either way, there appears to be no relationship between `B.ord` and `Singular.ordering(base_ring(B.S))` other than coincidence if I start with some ideal created in OSCAR. So, if the `if` condition is rarely `true` (somebody should confirm this) and testing it is expensive (as you say), I'd suggest to remove it.